overlord/snapshotstate/snapshotstate.go

```diff
@@ -69,6 +73,70 @@ func allActiveSnapNames(st *state.State) ([]string, error) {
 	return names, nil
 }
 
+func automaticSnapshotExpiration(st *state.State) time.Duration {
+	// TODO: get from config
+	return DefaultAutomaticSnapshotExpiration
+}
+
+// saveExpiration saves expiration date of the given snapshot set in the state.
+// The state needs to be locked by the caller.
+func saveExpiration(st *state.State, setID uint64, expiryTime time.Time) error {
+	// Expirations actually map to time.Time, use string here as we don't need parsed times here.
+	var expirations map[uint64]string
```

Comment:
Hrm, hrm, I get why we are doing this (to optimize the load/save to avoid a few cycles with string parsing). And yet I find this harder to read even with the comment. If we do it I would expand the comment a wee bit:
```
// "expirations" are actually `map[unint64]time.Time` but to speed-up reading the data we use map[uint64]string
// as this will avoid the parsing of the string to time.Time. Here we don't need the parsed time so that is fine.
```
But TBH I would prefer time.Time for readability (especially since it seems like the amount of snapshots we have is a couple of dozens at most?). It also means we don't need to manually format the time.Time to a string below (which is also a bit magic). Anyway, just my 0.02¢ - its fine if we keep it this way.

Reply:
This is now gone after changing to "snapshots" map & storing a simple struct with expiry as the only field, per Samuele's comment.